### PR TITLE
Doc. en merge-tree-settings

### PR DESCRIPTION
**Pull request by den-crane**
I hereby agree to the terms of the CLA available at: https://yandex.ru/legal/cla/?lang=en

Changelog category (leave one):
- Documentation (changelog entry is not required)

### CODE REVIEW

File: docs/en/operations/settings/merge-tree-settings.md [hunk 1]
```diff
@@ -0,0 +1,189 @@
...
+
+A large number of parts in a table reduces performance of ClickHouse queries and increases ClickHouse boot time. Most often this is a consequence of an incorrect design (mistakes when choosing a partitioning strategy - too small partitions).
+
+## replicated_deduplication_window {#replicated-deduplication-window}
+
+The number of most recently inserted blocks for which Zookeeper stores hashes to check for duplicates.
+
+Possible values:
+
+-   Any positive integer.
```
Comment (alesapin) [inline docs/en/operations/settings/merge-tree-settings.md:97]: Zero is also valid (disable deduplication).

File: docs/en/operations/settings/merge-tree-settings.md [hunk 2]
```diff
@@ -0,0 +1,189 @@
...
+
+Default value: 161061273600 (150 GB).
+
+The merge scheduler periodically analyzes the sizes and number of parts in partitions, and if there is enough free resources in the pool, it starts background merges. Merges occur until the total size of the source parts is less than `max_bytes_to_merge_at_max_space_in_pool`.
+
+Merges initiated by `optimize final` ignore `max_bytes_to_merge_at_max_space_in_pool` and merge parts only taking into account available resources (free disk's space) until one part remains in the partition.
```
Comment (alesapin) [inline docs/en/operations/settings/merge-tree-settings.md:150]: `> until one part remains in the partition`
`optimize final` will re-merge even a single part.

File: docs/en/operations/settings/merge-tree-settings.md [hunk 3]
```diff
@@ -0,0 +1,189 @@
...
+The maximum total part size (in bytes) to be merged into one part, with the minimum available resources in the background pool.
+
+Possible values:
+
+-   Any positive integer.
+
+Default value: 1048576 (1 MB)
+
+`max_bytes_to_merge_at_min_space_in_pool` defines the maximum total size of parts which can be merged despite the lack of available disk space (in pool). This is necessary to reduce the number of small parts and the chance of `Too many parts` errors.
+Merges book disk space by doubling the total merged parts sizes. Thus, with a small amount of free disk space, a situation may happen that there is free space, but this space is already booked by ongoing merges, so other merges unable to start, and the number of small parts grows with every insert.
```
Comment (alesapin) [inline docs/en/operations/settings/merge-tree-settings.md:163]: `> space is already booked by ongoing merges`
ongoing _large_ merges?

File: docs/en/operations/settings/merge-tree-settings.md [hunk 4]
```diff
@@ -0,0 +1,189 @@
...
+
+For example if a partition has 299 active parts and parts_to_throw_insert = 300, parts_to_delay_insert = 150, max_delay_to_insert = 1, `INSERT` is delayed for `pow( 1 * 1000, (1 + 299 - 150) / (300 - 150) ) = 1000` milliseconds.
+
+## max_parts_in_total {#max-parts-in-total}
+
+If the total number of active parts in all partitions of a table exceeds the `max_parts_in_total` value `INSERT` is interrupted with the `Too many parts (N)` exception.
+
+Possible values:
+
+-   Positive integer.
```
Comment (alesapin) [inline docs/en/operations/settings/merge-tree-settings.md:85]: `Any`? To be consistent with the settings below.

File: docs/en/operations/settings/merge-tree-settings.md [hunk 5]
```diff
@@ -0,0 +1,189 @@
...
+
+The number of most recently inserted blocks for which Zookeeper stores hashes to check for duplicates.
+
+Possible values:
+
+-   Any positive integer.
+
+Default value: 100.
+
+The `Insert` command creates one or more blocks (parts). When inserting into Replicated tables, ClickHouse for [insert deduplication](../../engines/table-engines/mergetree-family/replication/) writes the hash-sums of the created parts into Zookeeper. Hash sums are stored only for the most recent `replicated_deduplication_window` blocks. The oldest hash sums are removed from Zookeeper.
```
Comment (alesapin) [inline docs/en/operations/settings/merge-tree-settings.md:101]:
```suggestion
The `Insert` command creates one or more blocks (parts). When inserting into Replicated tables, ClickHouse for [insert deduplication](../../engines/table-engines/mergetree-family/replication/) writes the hash sums of the created parts into Zookeeper. Hash sums are stored only for the most recent `replicated_deduplication_window` blocks. The oldest hash sums are removed from Zookeeper.
```

File: docs/en/operations/settings/merge-tree-settings.md [hunk 6]
```diff
@@ -0,0 +1,189 @@
...
+
+The number of seconds after which the hash sums of the inserted blocks are removed from Zookeeper.
+
+Possible values:
+
+-   Any positive integer.
+
+Default value: 604800 (1 week).
+
+Similar to [replicated_deduplication_window](#replicated-deduplication-window), `replicated_deduplication_window_seconds` specifies how long to store hash-sums of blocks for insert deduplication. Hashes older than `replicated_deduplication_window_seconds` are removed from Zookeeper, even if they are less than ` replicated_deduplication_window`.
```
Comment (alesapin) [inline docs/en/operations/settings/merge-tree-settings.md:115]:
```suggestion
Similar to [replicated_deduplication_window](#replicated-deduplication-window), `replicated_deduplication_window_seconds` specifies how long to store hash sums of blocks for insert deduplication. Hash sums older than `replicated_deduplication_window_seconds` are removed from Zookeeper, even if they are less than ` replicated_deduplication_window`.
```

File: docs/en/operations/settings/merge-tree-settings.md [hunk 7]
```diff
@@ -0,0 +1,189 @@
...
+
+-   Positive integer.
+
+Default value: 100000.
+
+A large number of parts in a table reduces performance of ClickHouse queries and increases ClickHouse boot time. Most often this is a consequence of an incorrect design (mistakes when choosing a partitioning strategy - too small partitions).
+
+## replicated_deduplication_window {#replicated-deduplication-window}
+
+The number of most recently inserted blocks for which Zookeeper stores hashes to check for duplicates.
```
Comment (alesapin) [inline docs/en/operations/settings/merge-tree-settings.md:93]:
```suggestion
The number of most recently inserted blocks for which Zookeeper stores hash sums to check for duplicates.
```

File: docs/en/operations/settings/merge-tree-settings.md [hunk 8]
```diff
@@ -0,0 +1,189 @@
...
+
+The number of rows that are read from the merged parts into memory.
+
+Possible values:
+
+-   Any positive integer.
+
+Default value: 8192
+
+Merge reads rows from parts in blocks of `merge_max_block_size` rows, then merges and writes the result into a new part. The read block is placed in RAM, so `merge_max_block_size` affects the size of the RAM required for the merge. Thus, merges can consume a large amount of RAM for tables with very wide rows (if the average row size is 100kb, then when merging 10 parts, (100kb * 10 * 8192) = ~ 8GB of RAM). By decreasing `merge_max_block_size`, you can reduce the amount of RAM required for a merge.
```
Comment (alesapin) [inline docs/en/operations/settings/merge-tree-settings.md:175]: `> you can reduce the amount of RAM required for a merge`
and slow down merge?

File: docs/en/operations/settings/merge-tree-settings.md (hunk 2 above)
Comment (den-crane) [reply docs/en/operations/settings/merge-tree-settings.md:151]: not relevant to a topic max_bytes_to_merge_at_max_space_in_pool

**Review by alesapin — APPROVED** [review]

**Pull request merged**